test/test_testing.py

```diff
@@ -816,7 +817,19 @@ def test_unexpected_error_compare(self, _):
         with self.assertRaisesRegex(RuntimeError, "unexpected exception"):
             torch.testing.assert_close(actual, expected)
 
+    @unittest.skipIf(not torch.backends.mps.is_available(), reason="MPS is not available.")
+    @unittest.expectedFailure
+    def test_mps_compat(self):
```

Comment:
Hi Philip, @albanD , can correct me if I am missing something. But currently the Testing infra is not setup to test MPS supported GPUs. We support Apple Silicon and AMD GPUs. The HW for testing Apple Silicon is running MacOS 11 (which won't be compatible with MPS support) and the AMD GPUs are currently not there.

We can definitely add these tests so that we can test them internally and make fixes if needed. Was that your intention ?

Reply:
The intention is to have a test that expectedly fails until MPS is properly supported by the comparison machinery. If that happens, the test will no longer fail and in turn the `@unittest.expectedFailure` decorator will turn it into an unexpected success failure. 

My hope is that whoever adds support for the last operation that is needed for the comparison also removes the compatibility code we currently have in the comparison machinery. This test enforces this.